crates/miden-tx/src/executor/mod.rs

```diff
@@ -182,6 +182,70 @@ impl TransactionExecutor {
             account_codes,
         )
     }
+
+    // SCRIPT EXECUTION
+    // --------------------------------------------------------------------------------------------
+
+    /// Executes an arbitrary script against the given account and returns the stack state at the
+    /// end of execution.
+    ///
+    ///
+    /// # Errors:
+    /// Returns an error if:
+    /// - If required data can not be fetched from the [DataStore].
+    /// - If the transaction host can not be created from the provided values.
+    /// - If the execution of the provided program fails.
+    #[maybe_async]
+    pub fn execute_program(
+        &self,
+        account_id: AccountId,
+        block_ref: BlockNumber,
+        source: &str,
+        assembler: Assembler,
+        advice_inputs: AdviceInputs,
+    ) -> Result<[Felt; 16], TransactionExecutorError> {
```

Comment:
Is there a reason not to use the interface I suggested in https://github.com/0xPolygonMiden/miden-base/issues/1161#issuecomment-2672767826? Basically:
```rust
pub fn execute_program(
    &self,
    account_id: AccountId,
    block_ref: BlockNumber,
    tx_script: TransactionScript,
    advice_inputs: AdviceInputs,
) -> Result<[Felt; 16], TransactionExecutorError>
```
This would also allow specifying inputs for the transaction script (i.e., values that get put on the top of the stack before the script is executed).

The user could then generate `TransactionScript` using `TransactionScript::compile()` method which is a bit more general than just using a string.

Reply:
Usage of the `TransactionScript` will indeed allow user to specify the transaction script inputs, but in that case creation of the final program will be more complicated. 

To use this procedure in testing we need to execute the prologue before the provided code. If the code is provided as a string, I can just modify it, adding the execution of the prologue right after the `begin`. But if we have already compiled `TransactionScript`, we need to have something like a `main` script, which will execute the prologue first and then call the code provided by the user. I can implement it, but I thought that for the testing purposes the first approach will be sufficient.

Reply:
> But if we have already compiled `TransactionScript`, we need to have something like a `main` script, which will execute the prologue first and then call the code provided by the user. I can implement it, but I thought that for the testing purposes the first approach will be sufficient.

This is not just for testing purposes (i.e., it needs to work in the client under regular release). So, we should probably create another `main` file and use it here.